src/easylink/step.py

```diff
@@ -1650,9 +1708,37 @@ def _configure_subgraph_steps(self) -> None:
         This method recursively traverses the ``StepGraph`` and sets the configuration
         state for each ``Step`` until reaching all leaf nodes.
         """
-        nodes = self._step.step_graph.nodes
-        for node in nodes:
-            step = nodes[node]["step"]
+        for node in self._nodes:
+            step = self._nodes[node]["step"]
             step.set_configuration_state(
-                self.pipeline_config, self.combined_implementations, self.input_data_config
+                self.parent_config, self.combined_implementations, self.input_data_config
+            )
+
+
+####################
+# Helper functions #
+####################
+
+
+def _validate_step_graph(
+    step_graph: StepGraph,
+    parent_config: LayeredConfigTree,
+    combined_implementations: LayeredConfigTree,
+    input_data_config: LayeredConfigTree,
+) -> dict[str, list[str]]:
```

Comment:
Add `TODO` to move this back to HierarchicalStep once ChoiceStep's `choices` are each a `Step` rather than a full graph.